File: bus-n-go-backend/src/main/kotlin/mk/ukim/finki/busngobackend/api/responses/CommuteResponse.kt
package mk.ukim.finki.busngobackend.api.responses

import mk.ukim.finki.busngobackend.domain.entities.PostojkaNaLinija
import java.time.LocalDateTime

data class CommuteResponse(
    val commuter: UserResponse,
    val ticket: TicketResponse,
    val stationStart: PostojkaNaLinija,
    val status: String,
    val startDate: LocalDateTime,
    val endDate: LocalDateTime?,
    val id: Long,
)

File: bus-n-go-frontend/src/app/model/responses/CommuteResponse.ts
import {TicketResponse} from "./TicketResponse";
import {UserResponse} from "./UserResponse";
import {LineStation} from "../LineStation";

export interface CommuteResponse {
commuter: UserResponse,
   ticket: TicketResponse,
   stationStart: LineStation,
   status: String,
   startDate: Date,
   endDate: Date | null,
   id: number,
}
